### PR TITLE
refactor test_general

### PR DESCRIPTION
**Pull request by lalon4**

### CODE REVIEW

File: discovery-infra/tests/test_general.py [hunk 1]
```diff
@@ -34,16 +35,17 @@ def test_cluster_unique_name(self, api_client, cluster):
...
 
+    @pytest.mark.regression
     def test_select_roles(self, api_client, cluster, nodes):
-        cluster_id = self.test_discovery(api_client, cluster, nodes)
-        self.set_host_roles(cluster_id=cluster_id, api_client=api_client)
-        hosts = api_client.get_cluster_hosts(cluster_id=cluster_id)
+        c = self.test_discovery(api_client, cluster, nodes)
+        c.set_host_roles()
+        hosts = c.get_hosts()
         for host in hosts:
```
Comment (nshidlin) [inline discovery-infra/tests/test_general.py:49]: you can use cluster.set_host_roles here

Comment (lalon4) [reply discovery-infra/tests/test_general.py:49]: but here we assert the role in between

Comment (nshidlin) [reply discovery-infra/tests/test_general.py:49]: right, missed that

**Review by YuviGold — APPROVED** [review]
/approve

**Pull request merged**